### PR TITLE
add disable attribute for tc:buttons and tc:links

### PR DESCRIPTION
**Pull request by JasminKroeger**
* add setter for disable attribute in ButtonsTagDeclaration and LinksTagDeclaration
* create Interface to set and get disabled
* adjust Renderer to write disabled attribute in tag
* adjust isDisabled in AbstractUICommandBase to check if the parent is disabled. If the parent is disabled all all children are disabled. If a child has the disabled attribute set, than the child is not disabled.
* create example in demo

ISSUE: TOBAGO-1997

### CODE REVIEW

File: tobago-core/src/main/java/org/apache/myfaces/tobago/component/SupportsDisabledProperty.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.myfaces.tobago.component;
+
+public interface SupportsDisabledProperty {
```
Comment (lofwyr14) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/component/SupportsDisabledProperty.java:22]: Naming thing: Other similiar interfaces doesn't have the "Property" suffix in the name. Please change to "SupportsDisabled".

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/component/AbstractUIButtons.java [hunk 2]
```diff
@@ -19,12 +19,28 @@
...
+public abstract class AbstractUIButtons extends AbstractUIPanelBase implements SupportsDisabled {
+
+  enum PropertyKeys {
+    disabled,
+  }
 
   public abstract Orientation getOrientation();
+
+  @Override
+  public boolean isDisabled() {
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/component/AbstractUIButtons.java:37]: I think getter/setter for disabled could be generated. So it's no need to implement it here.

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/component/AbstractUILinks.java [hunk 3]
```diff
@@ -19,12 +19,28 @@
...
+public abstract class AbstractUILinks extends AbstractUIPanelBase implements SupportsDisabled {
+
+  enum PropertyKeys {
+    disabled,
+  }
 
   public abstract Orientation getOrientation();
+
+  @Override
+  public boolean isDisabled() {
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/component/AbstractUILinks.java:37]: I think getter/setter for disabled could be generated. So it's no need to implement it here.

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/taglib/component/ButtonsTagDeclaration.java [hunk 4]
```diff
@@ -46,4 +48,10 @@
 public interface ButtonsTagDeclaration
     extends HasIdBindingAndRendered, IsVisual, HasTip, HasOrientation {
 
+  /**
+   * Flag indicating that this element and all children are disabled.
+   */
+  @TagAttribute()
+  @UIComponentTagAttribute(type = "boolean", generate = false)
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/taglib/component/ButtonsTagDeclaration.java:55]: I think getter/setter for disabled could be generated. Remove "generate = false".

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/taglib/component/LinksTagDeclaration.java [hunk 5]
```diff
@@ -43,4 +45,11 @@
     },
     rendererType = {RendererTypes.LINKS, RendererTypes.LINKS_INSIDE_BAR})
 public interface LinksTagDeclaration extends HasIdBindingAndRendered, IsVisual, HasTip, HasOrientation {
+
+  /**
+   * Flag indicating that this element and all children are disabled.
+   */
+  @TagAttribute()
+  @UIComponentTagAttribute(type = "boolean", generate = false)
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/taglib/component/LinksTagDeclaration.java:53]: I think getter/setter for disabled could be generated. Remove "generate = false".

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/ButtonsRenderer.java [hunk 6]
```diff
@@ -57,6 +57,9 @@ public void encodeBegin(final FacesContext facesContext, final UIComponent compo
             ? BootstrapClass.BTN_GROUP_VERTICAL : BootstrapClass.BTN_GROUP,
         buttons.getCustomClass());
     writer.writeAttribute(HtmlAttributes.ROLE, HtmlRoleValues.GROUP.toString(), false);
+    if (buttons.isDisabled()) {
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/ButtonsRenderer.java:60]: The HTML DIV tag includes the global attributes but no "disabled" attribute.
See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div

Imho the "disabled" attribute should not be rendered.

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/LinksRenderer.java [hunk 7]
```diff
@@ -48,6 +49,9 @@ public void encodeBegin(final FacesContext facesContext, final UIComponent compo
         getExtraCssItem(),
         Orientation.vertical.equals(links.getOrientation()) ? BootstrapClass.FLEX_COLUMN : null,
         links.getCustomClass());
+    if (links.isDisabled()) {
```
Comment (henningn) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/LinksRenderer.java:52]: The HTML DIV tag includes the global attributes but no "disabled" attribute.
See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div

Imho the "disabled" attribute should not be rendered.

**Pull request merged**